### PR TITLE
[SPARK-31581][R] Replaces paste(sep="") to paste0

### PR DESCRIPTION
**Pull request by MichaelChirico**
### What changes were proposed in this pull request?

All instances of `paste(..., sep = "")` in the code are replaced with `paste0` which is more performant

### Why are the changes needed?

Performance & consistency (`paste0` is already used extensively in the R package)

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

None

### CODE REVIEW

File: R/pkg/R/mllib_classification.R [hunk 1]
```diff
@@ -338,7 +338,7 @@ setMethod("spark.logit", signature(data = "SparkDataFrame", formula = "formula")
               if (!is.null(lowerBoundsOnCoefficients) && (row != nrow(upperBoundsOnCoefficients)
                 || col != ncol(upperBoundsOnCoefficients))) {
                 stop(paste0("dimension of upperBoundsOnCoefficients ",
-                           "is not the same as lowerBoundsOnCoefficients", sep = ""))
+                            "is not the same as lowerBoundsOnCoefficients"))
```
Comment (HyukjinKwon) [inline R/pkg/R/mllib_classification.R:341]: This space seems a mistake.

**Review by HyukjinKwon — APPROVED** [review]

File: R/pkg/R/mllib_classification.R (hunk 1 above)
Comment (MichaelChirico) [reply R/pkg/R/mllib_classification.R:341]: it's aligning to the `"` on the previous line. I can revert if you want to minimize the diff

Comment (HyukjinKwon) [reply R/pkg/R/mllib_classification.R:341]: oh, that's true. I misread.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by HyukjinKwon** [comment]
Oops, I missed that the title linked a wrong JIRA.

**Pull request closed without merging**